### PR TITLE
Add upsert tests

### PR DESCRIPTION
**Pull request by lilsweetcaligula**
Adds upsert tests to the Shared generic store plugin test suite.

### CODE REVIEW

File: lib/common.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+exports.fetchProp = function (o, prop) {
```
[the remark this answers is not in the document]
Comment (lilsweetcaligula) [reply lib/common.js:1]: @rjrodger That's correct. It's just a make-shift safety helper for use until we migrate to TypeScript.

[the remark this answers is not in the document]
Comment (rjrodger) [reply lib/common.js:1]: Please remove these

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "seneca-store-test",
-  "version": "3.1.0",
+  "version": "3.2.0",
```
Comment (rjrodger) [inline package.json:3]: don't bump version unless publishing

File: package.json [hunk 3]
```diff
@@ -58,6 +58,6 @@
     "coveralls": "^3.1.0",
     "seneca": "plugin",
     "seneca-entity": "^10.1.0",
-    "seneca-mem-store": "^4.0.0"
+    "seneca-mem-store": "5.0.0"
```
Comment (rjrodger) [inline package.json:61]: package.json deps are now inconsistent - either use ^ or change all to be fixed

File: store-test.js [hunk 4]
```diff
@@ -10,6 +10,7 @@ const Code = require('@hapi/code')
 const Nid = require('nid')
 
 var ExtendedTests = require('./lib/store-test-extended')
+const { fetchProp } = require('./lib/common')
```
Comment (rjrodger) [inline store-test.js:13]: remove

File: store-test.js [hunk 5]
```diff
@@ -470,15 +492,15 @@ function basictest(settings) {
               return done(err)
             }
 
-            Assert.notOk(foo2.p1)
-            Assert.ok(foo2.p2)
+            expect(foo2.p1).to.equal(null)
```
Comment (rjrodger) [inline store-test.js:495]: good - we need to be consistent

File: store-test.js [hunk 6]
```diff
@@ -868,7 +890,7 @@ function basictest(settings) {
         var foo = si.make('foo')
         foo.native$(
           verify(done, function (driver) {
-            Assert.isObject(driver)
+            Assert.ok(driver)
```
Comment (rjrodger) [inline store-test.js:893]: use expect(...).exists()

File: store-test.js [hunk 7]
```diff
@@ -1319,6 +1341,1108 @@ function limitstest(settings) {
...
+  describe('Upserts', () => {
+    beforeEach(() => new Promise((resolve, _reject) => {
+      si.ready(resolve)
+    }))
+
+    beforeEach(clearDb(si))
+
+    afterEach(clearDb(si))
+
+    describe('save$ invoked on a new entity instance', () => {
```
Comment (rjrodger) [inline store-test.js:1362]: the first test should be a basic test of the core functionality- the "happy" path:
create an entity, do an upsert, verify data changed, and no duplicate

**Pull request merged**